### PR TITLE
Added advanced error messages

### PR DESCRIPTION
**Pull request by Henningstone**
This should be a bit more helpful than the current error messages...

### CODE REVIEW

**Comment by Henningstone** [comment]
replaced by #79 since it is an trivial feature

**Comment by Henningstone** [comment]
reopened since #79 cannot be merged

**Comment by petterarvidsson** [comment]
@Henningstone Can you change this to merge into master/HEAD? I'll be happy to merge it then!

File: src/client.c [hunk 1]
```diff
@@ -64,11 +64,11 @@ void client_send(char *data) {
     int length = strlen(data);
     int header_size = htonl(length);
     if (client_sendall(sd, (char*)&header_size, sizeof(header_size)) == -1) {
-        perror("client_sendall");
+        SHOWERROR("client_sendall");
         exit(1);
     }
     if (client_sendall(sd, data, length) == -1) {
-        perror("client_sendall");
+		SHOWERROR("client_sendall");
```
Comment (nsg) [inline src/client.c]: Please, change the tabs to spaces. The code will be hard to read with mixed indentation levels.

Comment (Henningstone) [inline src/client.c]: I'm sorry, that was Visual Studio. I'll fix that and commit to master again.

**Pull request closed without merging**